### DISCUSSION TITLE
How to create custom data type in TypeScript?

### DISCUSSION THREAD

**Discussion by oleksandr-petryk** [sequelize/sequelize#14364] [Help & Questions] [answered]
Recently I tried to create my own data type based on this [example](https://sequelize.org/docs/v6/other-topics/extending-data-types/), I did it on JS, but I didn't succeed. Now I use TypeScript, how I can make it?

I try to create data type to store only time with timezone #14213

**Comment by ephys** [comment]
Honestly it's probably very hard to write them in TypeScript right now. It's best to wait until their TypeScript migration has been completed https://github.com/sequelize/sequelize/pull/13799

I'll try to take a proper look at https://github.com/sequelize/sequelize/pull/14213

**Comment by ephys** [comment, marked as answer]
Sequelize 7 is typescript-friendly and the documentation on custom data types has been updated: https://sequelize.org/docs/v7/other-topics/extending-data-types/
